File: stack&queue/leetcode_1614/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var maxDepth = function(s) {
    let peak = 0;
    let balance = 0;

    for(const char of s) {
        if(char === '(') {
            balance ++;

            if(balance > peak) {
                peak ++;
            }
        } else if(char === ')') {
            balance --;
        }
    }

    return peak;
};

File: stack&queue/leetcode_1614/solution.py
class Solution:
    def maxDepth(self, s: str) -> int:
        peak = 0
        balance = 0

        for char in s:
            if char == '(':
                balance += 1

                if balance > peak:
                    peak += 1

            elif char == ')':
                balance -= 1

        return peak 